src/checkbox/checkbox.js

```diff
@@ -15,255 +15,214 @@
-   * https://github.com/jasonmayes/mdl-component-design-pattern
+   * Initialize checkbox from a DOM node.
    *
-   * @constructor
-   * @param {HTMLElement} element The element that will be upgraded.
-   */
-  var MaterialCheckbox = function MaterialCheckbox(element) {
-    this.element_ = element;
+   * @param {Element} root The element being upgraded.
+   */
+  constructor(root) {
+    // Check if the root has the right class.
+    this.root_ = root;
+    if (!root.classList.contains(MaterialCheckbox.classes_.ROOT)) {
+      throw new Error(
+          `MaterialCheckbox missing ${MaterialCheckbox.classes_.ROOT} class.`);
+    }
 
-    // Initialize instance.
-    this.init();
-  };
-  window['MaterialCheckbox'] = MaterialCheckbox;
+    // Look for required sub-nodes in the root's DOM.
+    this.input_ = root.querySelector(`.${MaterialCheckbox.classes_.INPUT}`);
+    if (!this.input_) {
+      throw new Error(
+          `MaterialCheckbox missing ${MaterialCheckbox.classes_.INPUT} node.`);
+    }
 
-  /**
-   * Store constants in one place so they can be updated easily.
-   *
-   * @enum {string | number}
-   * @private
-   */
-  MaterialCheckbox.prototype.Constant_ = {
-    TINY_TIMEOUT: 0.001
-  };
+    // Initialize event listeners.
+    this.changeListener_ = this.refresh.bind(this);
+    this.focusListener_ =
+        () => this.root_.classList.add(MaterialCheckbox.classes_.IS_FOCUSED);
+    this.blurListener_ =
+        () => this.root_.classList.remove(MaterialCheckbox.classes_.IS_FOCUSED);
+    this.mouseUpListener_ = this.blur_.bind(this);
+
+    // Attach event listeners to the DOM.
+    this.addEventListeners();
+
+    // Refresh component.
+    this.refresh();
+  }
 
   /**
-   * Store strings for class names defined by this component that are used in
-   * JavaScript. This allows us to simply change it in one place should we
-   * decide to modify at a later date.
+   * String constants used in this component.
    *
-   * @enum {string}
    * @private
    */
-  MaterialCheckbox.prototype.CssClasses_ = {
-    INPUT: 'mdl-checkbox__input',
-    BOX_OUTLINE: 'mdl-checkbox__box-outline',
-    FOCUS_HELPER: 'mdl-checkbox__focus-helper',
-    TICK_OUTLINE: 'mdl-checkbox__tick-outline',
-    RIPPLE_EFFECT: 'mdl-js-ripple-effect',
-    RIPPLE_IGNORE_EVENTS: 'mdl-js-ripple-effect--ignore-events',
-    RIPPLE_CONTAINER: 'mdl-checkbox__ripple-container',
-    RIPPLE_CENTER: 'mdl-ripple--center',
-    RIPPLE: 'mdl-ripple',
-    IS_FOCUSED: 'is-focused',
-    IS_DISABLED: 'is-disabled',
-    IS_CHECKED: 'is-checked',
-    IS_UPGRADED: 'is-upgraded'
-  };
+  static get strings_() {
+    return {
+      CLASS_NAME: 'MaterialCheckbox'
+    };
+  }
 
   /**
-   * Handle change of state.
+   * CSS classes used in this component.
    *
-   * @param {Event} event The event that fired.
    * @private
    */
-  MaterialCheckbox.prototype.onChange_ = function(event) {
-    this.updateClasses_();
-  };
+  static get classes_() {
+    return {
```

Comment:
I take everything back anyways. My intention was to be an easy target for microbenchmarkers. Turns out, my version is slower in a microbenchmark :joy: 

In case you are interested: http://codepen.io/surma/pen/YqKMex?editors=0010